config.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Flask config."""
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+class Config:
+    """Base config."""
+
+    FLASK_ENV = 'production'
+    TESTING = False
+    DEBUG = False
+    
+    PHILIPS_HUE_URL = os.environ.get('PHILIPS_URL')
+    
+    
+class ProdConfig(Config):
+    FLASK_ENV = 'production'
```

Comment:
This is still doing the same thing as earlier.

Config is the base class.

ProdConfig inherits from config.

So ProdConfig has all of the variables of Config, plus any new variables defined in ProdConfig. If a variable is defined in Config and ProdConfig, the ProdConfig variable takes precedence (it overrides the Config variable).

So the FLASK_ENV = 'production' in ProdConfig is a no-op, since it already inherited that value from Config.

This can definitely be confusing, since implementation inheritance is python is fairly implicit (E.g. nowhere in your base class do you specify an interface, and no where in your inherited class do you specify what the parent interface is required to provide, or which variable your subclass is overriding itself).

Also note that, in general, implementation inheritance is discouraged (google 'composition over inheritance' or 'avoid implementation inheritance'). However, considering that this seems to be a widely adopted Flask pattern, it's probably good just to be conventional. I just bring this last point up because, at least in Python, we already found some of the pitfalls of implementation inheritance.

Reply:
Also note that you can write:

class ProdConfig(Config):
      pass

To indicate to the interpreter that you plan to implement the class later (to keep it from complaining).

Reply:
Ah sorry that makes sense. You're definitely right, and I think using the base config as the production config messed with my intuition a bit which was why in my head I felt that leaving this line made sense, but since it's being inherited I see that it's redundant.